src/fin/fin_interface.cpp

```diff
@@ -0,0 +1,592 @@
+    case 76: SetObject<miopen::solver::conv::ConvMPBidirectWinograd_xdlops<3, 3>>(); break;
+    case 77: SetObject<miopen::solver::conv::ConvMPBidirectWinograd_xdlops<4, 3>>(); break;
+    case 78: SetObject<miopen::solver::conv::ConvMPBidirectWinograd_xdlops<5, 3>>(); break;
+    case 79: SetObject<miopen::solver::conv::ConvMPBidirectWinograd_xdlops<6, 3>>(); break;
+    case 80: SetObject<miopen::solver::conv::ConvHipImplicitGemmForwardV4R5Xdlops>(); break;
+    case 81:
+        SetObject<miopen::solver::conv::ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm>();
+        break;
+    case 83: SetObject<miopen::solver::conv::ConvHipImplicitGemmWrwV4R4Xdlops_Padded_Gemm>(); break;
+    case 98: SetObject<miopen::solver::conv::ConvMlirIgemmFwd>(); break;
+    case 99: SetObject<miopen::solver::conv::ConvMlirIgemmBwd>(); break;
+    case 100: SetObject<miopen::solver::conv::ConvMlirIgemmWrW>(); break;
+    case 103: SetObject<miopen::solver::conv::ConvMlirIgemmFwdXdlops>(); break;
+    case 104: SetObject<miopen::solver::conv::ConvMlirIgemmBwdXdlops>(); break;
+    case 105: SetObject<miopen::solver::conv::ConvMlirIgemmWrWXdlops>(); break;
+    case 107: SetObject<miopen::solver::conv::ConvAsmImplicitGemmGTCDynamicFwdXdlopsNHWC>(); break;
+    case 108: SetObject<miopen::solver::conv::ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC>(); break;
+    case 110: SetObject<miopen::solver::conv::ConvAsmImplicitGemmGTCDynamicWrwXdlopsNHWC>(); break;
+    case 114: SetObject<miopen::solver::conv::ConvCkIgemmFwdV6r1DlopsNchw>(); break;
+    case 127: SetObject<miopen::solver::conv::ConvAsmImplicitGemmGTCDynamicFwdDlopsNCHWC>(); break;
+    case 128: SetObject<miopen::solver::conv::ConvHipImplicitGemmFwdXdlops>(); break;
+    case 129: SetObject<miopen::solver::conv::ConvHipImplicitGemmBwdXdlops>(); break;
+    case 137: SetObject<miopen::solver::conv::ConvHipImplicitGemmGroupFwdXdlops>(); break;
+    case 138: SetObject<miopen::solver::conv::ConvHipImplicitGemm3DGroupFwdXdlops>(); break;
+    case 140: SetObject<miopen::solver::conv::ConvHipImplicitGemm3DGroupWrwXdlops>(); break;
+    case 141: SetObject<miopen::solver::conv::ConvHipImplicitGemm3DGroupBwdXdlops>(); break;
+    case 149: SetObject<miopen::solver::conv::ConvHipImplicitGemmF16F8F16FwdXdlops>(); break;
+    case 150: SetObject<miopen::solver::conv::ConvHipImplicitGemmF16F8F16BwdXdlops>(); break;
+    case 151: SetObject<miopen::solver::conv::ConvHipImplicitGemmF16F8F16WrwXdlops>(); break;
+    case 155: SetObject<miopen::solver::conv::ConvHipImplicitGemmGroupBwdXdlops>(); break;
+    case 156: SetObject<miopen::solver::conv::ConvHipImplicitGemmGroupWrwXdlops>(); break;
+    // New tunable solver should be added here
+    default: MIOPEN_THROW(miopenStatusInternalError, "Unknown solver ID");
+    }
+}
+
+template <>
+AnySolver<miopen::ExecutionContext, miopen::batchnorm::ProblemDescription>::AnySolver(
+    const miopen::solver::SolverBase* solver_base, uint64_t id)
+    : sbase(solver_base)
+{
```

Comment:
Probably it worth to keep each solver-specific part in a separate file. Otherwise, it can become really huge.
Like
```
fin_interface.cpp
fin_conv.cpp
fin_batchnorm.cpp
...etc...
```
It's just a recommendation. Or we can even keep it somewhere around the other solver-related files, like `solver/conv/fin_conv.cpp`

Reply:
Good point. Right now, it's not an issue, but it's worth thinking about how to ensure it doesn't turn into a mess in the future.